### PR TITLE
[maca] support deepseekv2 for maca backend.

### PR DESCRIPTION
**Pull request by Reinerzhou**

### CODE REVIEW

File: dlinfer/vendor/maca/maca_ops.py [hunk 1]
```diff
@@ -242,8 +253,7 @@ def paged_decode_attention(
         1,  # blocksparse_block_size
         1,  # blocksparse_head_sliding_step
     )
-
-    return output
+    return output[..., :512]
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/maca_ops.py:256]: 512 is only for MLA

File: dlinfer/vendor/maca/maca_ops.py [hunk 2]
```diff
@@ -13,6 +13,10 @@
 from dlinfer.utils.type_annotation import Tensor, Optional, Sequence, Tuple
 
 from .maca_extension import ops as maca_ext_ops
+from .context_flashattention import (
+    context_attention_fwd as context_attention_fwd_deepseek,
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/maca_ops.py:17]: rename it to mla, not deepseek

File: dlinfer/vendor/maca/maca_ops.py [hunk 3]
```diff
@@ -148,7 +153,7 @@ def prefill_attention(
         attn_output = attn_output.transpose(1, 2).flatten(0, 1)
         attn_output = attn_output[..., :nope_size].contiguous()
         attn_output = attn_output.to(input_type)
-        return attn_output[..., :512].contiguous()
+        return attn_output
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/maca_ops.py:156]: after padding, what is the attn_output.shape(-1)? 512or576?

File: dlinfer/vendor/maca/context_flashattention.py [hunk 4]
```diff
@@ -0,0 +1,731 @@
+# The kernels in this file are adapted from LightLLM's context_attention_fwd:
+# https://github.com/ModelTC/lightllm/blob/main/lightllm/models/llama/triton_kernel/context_flashattention_nopad.py
+
+import torch
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/context_flashattention.py:4]: move vendor kernels to dlinfer

**Review by jinminxi104 — CHANGES_REQUESTED** [review]

File: dlinfer/vendor/maca/maca_ops.py (hunk 2 above)
Comment (Reinerzhou) [reply dlinfer/vendor/maca/maca_ops.py]: done.

File: dlinfer/vendor/maca/maca_ops.py (hunk 1 above)
Comment (Reinerzhou) [reply dlinfer/vendor/maca/maca_ops.py]: done.
